routing.go

```diff
@@ -51,7 +49,7 @@ type routelet struct {
 // lifetime of the routelet is bound to the provided context. When the context is
 // cancelled, the routelet stops tracking the latest routing information.
 func newRoutelet(ctx context.Context, env env, group string) *routelet {
-	r := &routelet{env: env, balancers: map[string]*routingBalancer{}}
+	r := &routelet{balancers: map[string]*routingBalancer{}}
 	go r.watchRoutingInfo(ctx, env, group)
```

Comment:
I've noticed there are a couple places where goroutines are started in the `new` function for the type. I usually try to make concurrency as explicit as possible and have the caller start the goroutine by calling a `run` function (ex `go routelet.Run(ctx)`). That pattern requires the caller to do more work, but I think generally makes it easier to wait for goroutines to shutdown, because the caller can manage them using `sync.WaitGroup` or` errgroup.Group`.

I'm wondering if that pattern might help us solve the TODO below, or if it's not worth the change.

Reply:
Sounds like a great idea to me! Feel free to do in this PR or in a separate PR, whatever you prefer.